src/org/ods/openshift/OpenShiftResourceMetadata.groovy

```diff
@@ -0,0 +1,462 @@
+package org.ods.openshift
+
+
+import org.ods.services.OpenShiftService
+import org.ods.util.ILogger
+import org.ods.util.IPipelineSteps
+
+/**
+ * Utility class to handle recommended and custom labels and annotations for OpenShift resources.
+ *
+ * @See <a href="https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/" >
+ *     Kubernetes: Recommended Labels</a>
+ * @See <a href="https://github.com/gorkem/app-labels/blob/master/labels-annotation-for-openshift.adoc" >
+ *     Guidelines for Labels and Annotations for OpenShift applications</a>
+ * @See <a href="https://helm.sh/docs/chart_best_practices/labels/" > Helm: Labels and Annotations</a>
+ *
+ */
+class OpenShiftResourceMetadata {
```

Comment:
Please add `@TypeChecked`, that is really helpful.

Reply:
Will do.

Reply:
Done